Sources/Sentry/SentryHttpTransport.m

```diff
@@ -64,93 +62,12 @@ - (void)sendEnvelope:(SentryEnvelope *)envelope
         return;
     }
 
-    NSError *requestError = nil;
-    // TODO: We do multiple serializations here, we can improve this
-    NSURLRequest *request = [self createEnvelopeRequest:envelope didFailWithError:requestError];
-
-    if (nil != requestError) {
-        [SentryLog logWithMessage:requestError.localizedDescription andLevel:kSentryLogLevelError];
-        return;
-    }
-
-    // TODO: We do multiple serializations here, we can improve this
-    NSString *storedEnvelopePath = [self.fileManager storeEnvelope:envelope];
-
-    [self sendRequest:request storedPath:storedEnvelopePath];
+    [self.fileManager storeEnvelope:envelope];
+    [self sendAllCachedEnvelopes];
```

Comment:
Given that we must serialize/write an envelope to disk before we do anything (since an app can crash at anytime), and we only delete a file once we've managed to send (or try) to Sentry.

Assuming a sunny day scenario where the device if online, currently it works like:

1. Capture A is called
2. A is written to disk
3. `sendAllCachedEnvelopes` is called (ARC for A is 0 so memory reclaimed)
4. File is read from disk (it's A)
5. A is deserialized
6. A is serialized to be put into the network socket
7. A is deleted.

What about we try to optimize this a little:

1. Capture A is called
2. A is written to disk.
3. `sendAllCachedEnvelopes` is called (with A as a parameter)
4. File is read from disk
5. File name is compared (match!) with reference to object A
6. A is serialized to be put into the network socket
7. A is deleted.

This assumes `sendAllCachedEnvelopes`'s `for` loop cycles through file names and avoid reading the file altogether (also then  deserializing any bytes).

So we'd assume the read I/O of the file and deserializing it into an object by comparing the reference we already had in memory with the current file we're processing from disk.

Reply:
Just to be sure we are on the same page. You want to change the logic like you suggested to save one serialization right?
Because before this PR we had two serializations: one for creating the request, and one for saving to disk. Now we have three: one for saving it to the disk, one for reading it from the disk, and one for creating the request.

I can try to improve this yep.

Reply:
I now save one serialization by directly passing the data to `createEnvelopeRequest` instead of serializing it again which makes sending envelopes around 10% faster.

Avoiding to read the file altogether would need changes in the `SentryFileManager` and I wish not to do this in this PR. I think we could also do some performance improvements in there.

Reply:
yeah a follow up PR is fine. i was hoping the loop goes through "file names" only and not actually references that already hold the blob in memory. We could read the bytes lazily like `getBytes` or whatever.